modules/loaders-common/src/test/scala/com.snowplowanalytics.snowplow.loaders/transform/TransformSpec.scala

```diff
@@ -245,4 +268,93 @@ object TransformSpec {
   val testSchemaKey702 = SchemaKey.fromUri("iglu:com.example/mySchema/jsonschema/7-0-2").toOption.get
   val testSchemaKey801 = SchemaKey.fromUri("iglu:com.example/mySchema/jsonschema/8-0-1").toOption.get
+
+  val typeCaster = new Caster[String] {
+    override def nullValue: String                                             = "null"
+    override def jsonValue(v: Json): String                                    = "json"
+    override def stringValue(v: String): String                                = "string"
+    override def booleanValue(v: Boolean): String                              = "boolean"
+    override def intValue(v: Int): String                                      = "int"
+    override def longValue(v: Long): String                                    = "long"
+    override def doubleValue(v: Double): String                                = "double"
+    override def decimalValue(unscaled: BigInt, details: Type.Decimal): String = "double"
+    override def timestampValue(v: Instant): String                            = "timestamp"
+    override def dateValue(v: LocalDate): String                               = "date"
+    override def arrayValue(vs: List[String]): String                          = "array"
+    override def structValue(vs: List[Caster.NamedValue[String]]): String      = "struct"
+  }
+
+  val typeCirceFolder = new Json.Folder[String] {
+    override def onNull: String                       = "null"
+    override def onBoolean(value: Boolean): String    = "boolean"
+    override def onNumber(value: JsonNumber): String  = "number"
+    override def onString(value: String): String      = "string"
+    override def onArray(value: Vector[Json]): String = "array"
+    override def onObject(value: JsonObject): String  = "object"
+  }
+
+  def genEvents(n: Int): List[Event] =
+    (1 to n).toList.flatMap(_ =>
+      runGen(
+        GenSdkEvent.gen(1, 1, Instant.now, EventFrequencies(1, 1, 1, 1, 1, UnstructEventFrequencies(1, 1, 1))),
+        new Random()
+      )
+    )
+
+  def getParams(event: Event): Map[String, Any] = {
+    val values = event.productIterator.filter {
+      case _: Contexts | _: UnstructEvent => false
+      case _                              => true
+    }
+    // We can use 'Product.productElementNames' method to get field names
+    // however that method is introduced in Scala 2.13. Since we want to run
+    // tests with Scala 2.12 as well, we can't use that method.
+    (AtomicFields.static.map(_.name) zip values.toList).toMap +
+      ("contexts" -> event.contexts) +
+      ("unstruct_event" -> event.unstruct_event)
+  }
+
+  def getFieldValueFromEvent(eventMap: Map[String, Any], fieldName: String): Json =
+    if (fieldName.startsWith("contexts_"))
+      eventMap("contexts").asInstanceOf[Contexts].toShreddedJson(fieldName)
+    else if (fieldName.startsWith("unstruct_"))
+      eventMap("unstruct_event").asInstanceOf[UnstructEvent].toShreddedJson.get._2
+    else
```

Comment:
Is it really too much different ? 😄

Reply:
It might be a matter of taste but I think that this is good practice to use pattern matching whenever we can in Scala, as it makes it clear that all the conditions are for the same field, which is not the case with `else if`